Mathlib/Order/Hom/Set.lean

```diff
@@ -157,3 +157,97 @@ theorem compl_antitone : Antitone (compl : α → α) :=
 #align compl_antitone compl_antitone
 
 end BooleanAlgebra
+
+section Set
+
+variable {s t : Set α}
+
+/-- Equivalent types have order-isomorphic types of subsets. --/
+def Equiv.setOrderIso (e : α ≃ β) : Set α ≃o Set β where
```

Comment:
I think this is defeq abuse, and should actually be:

```suggestion
def Equiv.setOrderIso (e : α ≃ β) : (@HasSubset.subset (Set α) _) ≃r (@HasSubset.subset (Set β) _) where
```
Perhaps @YaelDillies has an opinion.

Reply:
This makes sense, and it does make some of the simp lemmas unneccesary. 

But it shows junk in the infoview; for example, `Function.Embedding.setOrderEmbedding f` shows as having type `(fun x x_1 ↦ x ⊆ x_1) ↪r fun x x_1 ↦ x ⊆ x_1` rather than the readable `α ↪o β`

Reply:
I think `Set α ≃o Set β`. However I'm pretty sure I recently added this definition to mathlib. Let me check.

Reply:
`Equiv.toOrderIsoSet`

Reply:
I'm slightly inclined to agree with Eric about the defeq abuse, but I don't feel strongly about it.

Reply:
Ah, I missed that this definition is a move/rename, so indeed this objection is out of scope even if it's valid.

I recall that we had some trouble during the port due to mixups between le and subset on order homs / isos; perhaps @ChrisHughes24 remembers more? (Or perhaps I'm misremembering who was involved in diagnosing it)